### DISCUSSION TITLE
Cloud run issue : ATDRunner always starting a local server

### DISCUSSION THREAD

**Discussion by AnilPatidar** [AppiumTestDistribution/AppiumTestDistribution#1086] [General]
@saikrishna321 sir, below ATDRunner code causing some issue when running on cloud platform like headspin etc.
Am i correct to say this?

    public ATDRunner() throws Exception {
        capabilities = Capabilities.getInstance();
        writeServiceConfig();
        AppiumServerManager appiumServerManager = new AppiumServerManager();
        appiumServerManager.startAppiumServer("127.0.0.1"); //Needs to be removed
        List<Device> devices = Devices.getConnectedDevices();
        ATDExecutor = new ATDExecutor(devices);
        createOutputDirectoryIfNotExist();
    }

**Comment by AnilPatidar** [comment]
@saikrishna321 sir could you please guide here ,if any code change would be required?

**Reply by saikrishna321** [reply]
No code changes! This is expected! ATD works with device-farm plugin which will proxy all request to cloud.

**Reply by Jitu1888** [reply]
How local Appium server get the cloud devices? @saikrishna321

**Reply by saikrishna321** [reply]
with appium 1.0 all the logic were inside ATD later with plugin capabilities in appium 2.0 we moved all the code logic to the server side to make it language neutral. In this change we if its remote, local or cloud execution everything is taken care by the plugin in terms on device orchestration and session mgt. 

So based on the configuration and every request we just route the command to the cloud.
